app/main/views/agreements.py

```diff
@@ -0,0 +1,71 @@
+from flask import jsonify, abort
+from .. import main
+from sqlalchemy.exc import IntegrityError
+from dmapiclient.audit import AuditTypes
+from ...models import (
+    db, FrameworkAgreement, AuditEvent
+)
+
+from ...utils import (
+    get_json_from_request,
+    json_has_required_keys,
+    json_has_keys,
+    validate_and_return_updater_request,
+)
+
+from ...supplier_utils import validate_agreement_details_data
+
+
+@main.route('/agreements/<int:agreement_id>', methods=['GET'])
+def get_framework_agreement(agreement_id):
+    framework_agreement = FrameworkAgreement.query.filter(FrameworkAgreement.id == agreement_id).first_or_404()
+
+    return jsonify(agreement=framework_agreement.serialize())
+
+
+@main.route('/agreements/<int:agreement_id>', methods=['POST'])
+def update_framework_agreement(agreement_id):
+    framework_agreement = FrameworkAgreement.query.filter(FrameworkAgreement.id == agreement_id).first_or_404()
+
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+    json_has_required_keys(json_payload, ["agreement"])
+    update_json = json_payload["agreement"]
+
+    json_has_keys(update_json, optional_keys=['signedAgreementDetails', 'signedAgreementPath'])
```

Comment:
It really is good to get into the habit of using tuples for sequences that are intended to be ephemeral.